src/Application.php

```diff
@@ -639,33 +468,15 @@ private function checkForDuplicateRoute($path, $methods = null)
      */
     private function emitMarshalServerRequestException($exception)
     {
-        $response = (new Response())
-            ->withStatus(StatusCode::STATUS_BAD_REQUEST);
-        $finalHandler = $this->getFinalHandler();
-        $response = $finalHandler(new ServerRequest(), $response, $exception);
+        if ($this->container && $this->container->has(Middleware\ErrorResponseGenerator::class)) {
```

Comment:
I understand that this is before the stack is formed, I was just hoping to not have two sets of error handling - one for "startup" exceptions, and one for when the stack is run. No listeners means no custom logging, notifications, etc.

What about something like:
```
if ($this->container && $this->container->has(Middleware\ErrorResponseGenerator::class)) {
    //This is factoried already. Can be modified by user to include listeners, etc.
    $errorHandler = $this->container->get(Middleware\ErrorHandler::class);
    $response =  $errorHandler->handleStartupException(
        $exception, 
        new ServerRequest());
} else {
    $response = $this->responsePrototype->withStatus(StatusCode::STATUS_BAD_REQUEST);
}
```

With the new function in `Zend\Stratigility\Middleware\ ErrorHandler` being:
```
public function handleStartupException($e, $request)
{
    set_error_handler($this->createErrorHandler());
    $response = $this->handleThrowable($e, $request);
    restore_error_handler();

    return $response;
}
```

So we are constructing the `ErrorHandler`, but bypassing the pipeline `process()` because we already know there is an Exception. This would allow the generator and listeners to do their things and return a response to the emitter while keeping everything for both startup and process errors using the same code.